go-controller/pkg/cni/helper_linux.go

```diff
@@ -365,17 +365,41 @@ func ConfigureOVS(ctx context.Context, namespace, podName, hostIfaceName string,
 	// Find and remove any existing OVS port with this iface-id. Pods can
 	// have multiple sandboxes if some are waiting for garbage collection,
 	// but only the latest one should have the iface-id set.
-	uuids, _ := ovsFind("Interface", "_uuid", "external-ids:iface-id="+ifaceID)
-	for _, uuid := range uuids {
-		if out, err := ovsExec("remove", "Interface", uuid, "external-ids", "iface-id"); err != nil {
-			klog.Warningf("Failed to clear stale OVS port %q iface-id %q: %v\n  %q", uuid, ifaceID, err, out)
+	names, _ := ovsFind("Interface", "name", "external-ids:iface-id="+ifaceID)
+	for _, name := range names {
+		if name == hostIfaceName {
+			// this may be result of restarting ovnkube-node, and it is trying to add the same VF representor to
+			// br-int for the same pod; do not delete port in this case.
+			continue
+		}
+		if out, err := ovsExec("--if-exists", "--with-iface", "del-port", "br-int", name); err != nil {
+			klog.Warningf("Failed to delete stale OVS port %q with iface-id %q from br-int: %v\n %q",
+				name, ifaceID, err, out)
+		}
+	}
+
+	// if the specified port was created for other Pod/NAD, return error
+	extIds, err := ovsFind("Interface", "external_ids", "name="+hostIfaceName)
```

Comment:
This is a good check. But I am wondering, how this can happen where an interface was taken by another Pod/NAD and we are given this interface?

Reply:
this is to find all the exsiting ovs interface that is associated with the current pod/NAD (except the requested one), and delete the OVS interface from br-int. As we should have one one interface associated with this pod/NAD.

It can happen if the connectionDetails (Rep interface) is updated for this pod/NAD, but at the time of update, ovnkube-node-dpu is not running, and when ovnkube-node-dpu restarts, we need to configure the new Rep  interface and delete the old ones.

Reply:
Thanks for the explanation. But a few more questions... Are we counting on the fact that outside of `ConfigureOVS()`, in the error case, we delete the VF Rep/Port? Shouldn't we try to cleanup here instead of returning with an error?

Reply:
sorry that my previous answer was for code at line368. I misunderstood what were you asking.

I don't really recall what happened when we added the check at line382, but I can guess the reason is the following case:

1. podA is assigned with vfA cmdAdd in dpu-host and configureOVS on dpu all completed
2. podA cmdDel happens, it unset podA's connectionDetails annotation
3. podB cmdAdd, and again is assigned with vfA, update its connectionDetails
4. at this time, on dpu, podB's watchDpu handler configures vfA to br-int (if without adding this check)
5. on dpu, podA's watchDPU handler found emptied out connectionDetails and delete vfA from br-int, note that even we have the sandbox/nadName check in delRepPort(), the race could happen after the check.

so it is best to just stop configureOVS for podB at step 4. 

and cleanup is not a good idea as we don't know if the interface is really now used by other Pod.

Reply:
Oh wow, thanks for explaining the possible race. That helped me understand it a lot more. I have question and not a comment for changing anything:
For your example race. Let's say we stop the `configureOVS()` for podB at step 4. We will get an error `OVS port %s was added for iface-id (%s), now readding it for (%s)` then vfA will be deleted from br-int. From this code:
```
	err = cni.ConfigureOVS(context.TODO(), pod.Namespace, pod.Name, vfRepName, ifInfo, dpuCD.SandboxId, getter)
	if err != nil {
		// Note(adrianc): we are lenient with cleanup in this method as pod is going to be retried anyway.
		_ = bnnc.delRepPort(vfRepName)
		return err
	}
```
 How does podB recover? I understand podA would be ok on step 5. Doesn't `watchDpu()` only trigger on specific events?

Reply:
podB will retry if its cmdAdd timed out.